server/routes/elasticsearch.ts

```diff
@@ -133,6 +136,87 @@ const getAliases = async (
   }
 };
 
+const createIndex = async (
+  req: Request,
+  h: ResponseToolkit,
+  callWithRequest: CallClusterWithRequest
+): Promise<ServerResponse<any>> => {
+  //@ts-ignore
+  const index = req.payload.indexConfig.index;
+  //@ts-ignore
+  const body = req.payload.indexConfig.body;
+  try {
+    await callWithRequest(req, 'indices.create', {
+      index: index,
+      body: body,
+    });
+  } catch (err) {
+    console.log('Anomaly detector - Unable to create index', err);
+    return { ok: false, error: err.message };
+  }
+  try {
+    const response: CatIndex[] = await callWithRequest(req, 'cat.indices', {
+      index,
+      format: 'json',
+      h: 'health,index',
+    });
+    return { ok: true, response: { indices: response } };
+  } catch (err) {
+    // In case no matching indices is found it throws an error.
+    if (
+      err.statusCode === 404 &&
+      get<string>(err, 'body.error.type', '') === 'index_not_found_exception'
+    ) {
+      return { ok: true, response: { indices: [] } };
+    }
+    console.log('Anomaly detector - Unable to get indices', err);
+    return { ok: false, error: err.message };
+  }
+};
+
+const bulk = async (
+  req: Request,
+  h: ResponseToolkit,
+  callWithRequest: CallClusterWithRequest
+): Promise<ServerResponse<GetAliasesResponse>> => {
+  //@ts-ignore
+  const body = req.payload.body;
+  try {
+    const response: any = await callWithRequest(req, 'bulk', {
+      body: body,
+    });
+    //@ts-ignore
+    return { ok: true, response: { response } };
+  } catch (err) {
+    console.log('Anomaly detector - Unable to perform bulk action', err);
+    return { ok: false, error: err.message };
+  }
+};
+
+const deleteIndex = async (
+  req: Request,
+  h: ResponseToolkit,
+  callWithRequest: CallClusterWithRequest
+): Promise<ServerResponse<any>> => {
+  //@ts-ignore
+  const index = req.payload.index;
+  console.log('index to delete: ', index);
+
+  try {
+    const response: any = await callWithRequest(req, 'indices.delete', {
+      index: index,
+    });
+    //@ts-ignore
+    return { ok: true, response: { response } };
+  } catch (err) {
+    console.log(
```

Comment:
if indexNotFound exception is caught, should we return OK as `true`?

Reply:
Do you mean if an `indexNotFoundException` is thrown in the delete api call, still return ok as `true`? I feel like this should be `false` since there was an error while trying to delete, let me know what you think

Reply:
I personally prefer returning `true` as this can make this delete API idempotent, but this is from perspective of API, not a blocker.

Reply:
Got it, will change to return true if an `index_not_found_exception` is returned when performing delete. Will also re-retrieve the indices if successful to properly update redux store.